src/generators/number/index.ts

```diff
@@ -1,16 +1,17 @@
 import {Chance} from 'chance';
 
 /**
- * Generates a random `number`, optionally between `lowerInclusive` and `upperExclusive`.
+ * Generates a random `number`, optionally between `lowerInclusive` and `upperExclusive` with `fixed` number of digits after the decimal.
  */
 const createNumberGenerator = (chance: Chance.Chance) => (
   lowerInclusive = -1000000,
   upperExclusive = 1000000,
+  fixed = 4,
 ): number =>
   chance.floating({
     min: lowerInclusive,
-    // chance.floating() returns 4 digits after the decimal at most, and its `max` is inclusive
-    max: upperExclusive - 0.0001,
+    max: upperExclusive,
+    fixed,
```

Comment:
As discussed one minute ago in person (lol), let's restore the subtraction, since it's needed to convert from an exclusive max to an inclusive max. But since this definitionally means we can't support a precision of more than the epsilon value we're using, let's add an assertion to guarantee we never run into broken/undefined behavior.

So, something like (copied assertion from https://github.com/goodeggs/fake-eggs/blob/873190403d1b34b37d297add3505d8adf14534d9/src/generators/day/index.ts#L46-L48):

```typescript
/* ... */
     const EPSILON = 0.0000001;
     const MAX_PRECISION = 6;
/* ... */

  if (upperExclusive != null && upperExclusive > MAX_PRECISION) {
    throw new TypeError(`"upperExclusive" must be less than ${MAX_PRECISION} or null`);
  }

/* ... */
     max: upperExclusive - EPSILON,
/* ... */
```

Reply:
Addressed in 856e035